### PR TITLE
Add the RegisterCustomTemplate call

### PR DESCRIPTION
**Pull request by mcorbin**

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -13,15 +13,18 @@ type Template struct {
 	Format                string            `json:"format,omitempty" doc:"the format of the template."`
 	HostID                *UUID             `json:"hostid,omitempty" doc:"the ID of the secondary storage host for the template"`
 	HostName              string            `json:"hostname,omitempty" doc:"the name of the secondary storage host for the template"`
+	Hypervisor            string            `json:"hypervisor,omitempty" doc:"the target hypervisor for the template"`
```
Comment (greut) [inline templates.go]: wasn't this root only?

File: templates_test.go [hunk 2]
```diff
@@ -33,3 +33,9 @@ func TestDeleteTemplate(t *testing.T) {
 	_ = req.Response().(*AsyncJobResult)
 	_ = req.AsyncResponse().(*BooleanResponse)
 }
+
+func TestRegisterCustomTemplate(t *testing.T) {
+	req := &RegisterCustomTemplate{}
+	_ = req.Response().(*AsyncJobResult)
+	_ = req.AsyncResponse().(*[]Template)
```
Comment (greut) [inline templates_test.go]: pointer to slices are usually overkill.

File: templates.go (hunk 1 above)
Comment (mcorbin) [reply templates.go]: Nope, this field is also visible for non-root users.

**Comment by pierre-emmanuelJ** [comment]
👍

**Pull request merged**